lib/str-format.c

```diff
@@ -303,8 +305,8 @@ scan_positive_int(const gchar **buf, gint *left, gint field_width, gint *num)
     {
       if ((**buf) >= '0' && (**buf) <= '9')
         result = result * 10 + ((**buf) - '0');
-      else if (!isspace((int) **buf))
-        return FALSE;
+      else if (!isspace((int) **buf) || result != 0)
```

Comment:
Would it be readable to skip spaces first, then error in every nondigit letters? I do not understand how is this works. What is the connection between terminating spaces and result? We want to accept `" 0 "`, and only ignore `" 3 "`?

Reply:
I actually tried unrolling the loop and skip initial spaces, but the code got slower in my microbenchmark and not by a bit. We can do ~300m timestamp parsing per second and with unrolling it went down to ~280m

I guess the compiler is too damn smart about optimizing this loop, and it gets worse when I use two separate loops.

My intention was to accept initial whitespace but don't accept whitespace at the end of the field, but you are right that this accepts " 0 ", which is not good. (I was using the "result != 0" condition to see if we already processed digit characters, not wanting to introduce a separate variable for that fact).

Let me see what I can do here.

Reply:
I improved it, made some measurements again and I think it got better. Thanks for pushing this.

Reply:
Looks nice, thank you